packages/keyring-core/test/agent.ts

```diff
@@ -1,29 +0,0 @@
-import test from 'ava'
-import { JSDOM } from 'jsdom'
-import 'fake-indexeddb/auto'
-
-import { createAgent } from '../src/index'
-
-test.before((t) => {
-  const dom = new JSDOM('<!DOCTYPE html>')
-  globalThis.document = dom.window.document
```

Comment:
how did this work with the `crypto` dependency previously?

Reply:
I'm not sure what the difference was - it's especially confusing that it was working with my stale node_modules...

The error was coming from this bit in `@noble/ed25519`:

```js
    sha512: async (...messages) => {
        const message = concatBytes(...messages);
        if (crypto.web) {
            const buffer = await crypto.web.subtle.digest('SHA-512', message.buffer);
            return new Uint8Array(buffer);
        }
        else if (crypto.node) {
            return Uint8Array.from(crypto.node.createHash('sha512').update(message).digest());
        }
        else {
            throw new Error("The environment doesn't have sha512 function");
        }
    },
```

And the error message was that `digest` was undefined. Which is a bit odd, since it suggests that `crypto.web` was defined, but `crytpo.web.subtle.digest` was not... Best guess is that the vitest jsdom env explicitly sets the `crypto` var to some non-empty value that's not actually backed by the crypto API. But that seems like a weird thing to do, so maybe not...

Reply:
I added some console logs to the vitest plugin to try to figure out what's going on. Seems like they define a `crypto` global with just the `getRandomValues` function defined...

Here's my logging code:

```js
    console.log('!!! old crypto global:')
    console.dir(global.crypto)
    for (const p in global.crypto) {
      console.log(`crypto prop ${p}: ${global.crypto[p]}`)
    }
```

And the output:

```
!!! old crypto global:
Crypto {}
crypto prop getRandomValues: getRandomValues(array) {
      const esValue = this !== null && this !== undefined ? this : globalObject;
      if (!exports.is(esValue)) {
        throw new globalObject.TypeError(
          "'getRandomValues' called on an object that is not a valid instance of Crypto."
        );
      }

      if (arguments.length < 1) {
        throw new globalObject.TypeError(
          `Failed to execute 'getRandomValues' on 'Crypto': 1 argument required, but only ${arguments.length} present.`
        );
      }
      const args = [];
      {
        let curArg = arguments[0];
        if (ArrayBuffer.isView(curArg)) {
        } else {
          throw new globalObject.TypeError(
            "Failed to execute 'getRandomValues' on 'Crypto': parameter 1" + " is not of any supported type."
          );
        }
        args.push(curArg);
      }
      return esValue[implSymbol].getRandomValues(...args);
    }
```

Reply:
This seems to solve the mystery of why it worked with ava, although I still don't know why it was working with my stale node_modules. 

With ava, the `crypto` global was undefined, so `@noble/ed25519` took the "node crypto" branch. But with vitest we have just enough `crypto` to lead us down the webcrypto branch, but no `crypto.subtle` to seal the deal.